standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1839,13 +1839,27 @@ private MTable getMTable(String catName, String db, String table) {
     return nmtbl.mtbl;
   }
 
-  @Override
-  public List<Table> getTableObjectsByName(String catName, String db, List<String> tbl_names)
-      throws MetaException, UnknownDBException {
+  private void verifyDBExists(String catName, String db) throws UnknownDBException {
+    // Need to differentiate between an unmatched pattern and a non-existent database
+    Query dbExistsQuery = pm.newQuery(MDatabase.class, "name == db && catalogName == cat");
+    dbExistsQuery.declareParameters("java.lang.String db, java.lang.String cat");
+    dbExistsQuery.setUnique(true);
+    dbExistsQuery.setResult("name");
+    String dbNameIfExists = (String) dbExistsQuery.execute(db, catName);
+    if (org.apache.commons.lang3.StringUtils.isEmpty(dbNameIfExists)) {
+      throw new UnknownDBException("Could not find database " +
+              DatabaseName.getQualified(catName, db));
+    }
+    dbExistsQuery.closeAll();
+  }
+
+  private List<Table> getTableObjectsByNameUtil(String catName, String db, List<String> tbl_names,
```

Comment:
instead of adding getTableObjectsByNameUtil() method thats called by both flavors of getTableObjectsByName(), should we use
1) getTableObjectsByName(String catName, String db, List<String> tbl_names,
          GetProjectionsSpec projectionsSpec)

and then 
2) getTableObjectsByName(String catName, String db, List<String> tbl_names)
should invoke #1 with a null projectionSpec value